core/snapshots/storage/bolt.go

```diff
@@ -386,6 +386,12 @@ func CommitActive(ctx context.Context, key, name string, usage snapshots.Usage,
 		// Replace labels, do not inherit
 		si.Labels = base.Labels
 
+		// If the snapshot didn't have a parent when created, allow it
+		// to be rebased on a parent on commit.
+		if si.Parent == "" && si.Parent != base.Parent {
```

Comment:
Maybe the logic would be better as first checking `si.Parent != base.Parent`. If `si.Parent` is empty, set it, otherwise, we should probably throw an invalid argument error.